src/services/healthcheck.service.js

```diff
@@ -1,42 +1,96 @@
-import { config as hathorLibConfig, healthApi, txMiningApi } from '@hathor/wallet-lib';
+import { healthApi, txMiningApi } from '@hathor/wallet-lib';
+import { config as hathorLibConfig } from '@hathor/wallet-lib';
+import { Healthcheck, HealthcheckInternalComponent, HealthcheckHTTPComponent, HealthcheckCallbackResponse } from 'hathor-healthcheck-lib';
+import { initializedWallets } from './wallets.service';
 
-const { buildComponentHealthCheck } = require('../helpers/healthcheck.helper');
 const { friendlyWalletState } = require('../helpers/constants');
 
-const healthService = {
+class HealthService {
+  constructor(walletIds, includeFullnode, includeTxMiningService) {
+    this.healthcheck = new Healthcheck({
+      name: "hathor-wallet-headless",
+      warn_is_unhealthy: true
```

Comment:
question: Should this property be configurable?

Reply:
I think so. We support 3 possible statuses for healthchecks, which are `pass`, `fail` and `warn`.

However, we need to translate them into a 200 or 503 HTTP status code, because most infrastructure systems only consider the status code for healthchecks, not the response body.

So, some projects using this lib could need to consider `warn` as a 503, and some could prefer to consider it a 200 (in case they have some other way to parse the body, for instance)

What do you think?

Reply:
According to [the RFC we used as reference](https://datatracker.ietf.org/doc/html/draft-inadarei-api-health-check) for this, I understand the status for a `warn` in case we explicitly set it _not to be unhealthy_ should be `200`.

If a client application expects to receive a `503` on healthcheck warnings, the configuration for this headless instance should just be left as default - as it is right now with `warn_is_unhealthy: true`.

I can imagine some more complex use cases, but I think this boolean configuration with a default to `true` would suffice for almost all users.

Do you agree?

Reply:
Yes, I do. Done in https://github.com/HathorNetwork/hathor-wallet-headless/pull/376/commits/971d3b2df4220e04cfb9bed1b69b28a76f9f08ad